modules/images/webp-uploads/load.php

```diff
@@ -403,7 +403,29 @@ function webp_uploads_update_image_references( $content ) {
 	}
 
 	foreach ( $images as $img => $attachment_id ) {
-		$content = str_replace( $img, webp_uploads_img_tag_update_mime_type( $img, 'the_content', $attachment_id ), $content );
+
+		/**
+		 * Check for old uploaded images.
+		 */
+		if ( ! preg_match( '~src*=*["\'](.*?)["\']~', $img, $image_src ) ) {
+			continue;
+		}
+
+		if ( empty( $image_src ) ) {
+			continue;
+		}
+
+		/**
+		 * Get attachment id by the attachment url.
+		 */
+		$updated_attachment_id = attachment_url_to_postid( $image_src[1] );
+
+		if ( $attachment_id !== $updated_attachment_id ) {
+			$update_class_name = preg_replace( '/wp-image-([\d]+)/i', sprintf( 'wp-image-%d', $updated_attachment_id ), $img );
+			$content           = str_replace( $img, webp_uploads_img_tag_update_mime_type( $update_class_name, 'the_content', $updated_attachment_id ), $content );
+		} else {
+			$content = str_replace( $img, webp_uploads_img_tag_update_mime_type( $img, 'the_content', $attachment_id ), $content );
+		}
```

Comment:
In my setup it changes all old images to webp please check the attached screenshot.

We can't use $img variable `$img = preg_replace( '/wp-image-([\d]+)/i', sprintf( 'wp-image-%d', $attachment_id ), $img );` here because it will not replace anything in below statement. We must use different variable so it get replace.

![image](https://user-images.githubusercontent.com/10103365/170177238-64caca74-90cc-4a56-921b-8ceaa81abd41.png)

Reply:
@mukeshpanchal27 Thank you for the clarification. It happens sporadically for me. It works well for some posts and on one of the post, I see the `$updated_attachment_id` to be `0` for all images except for one. And because of this, the attachment metadata is empty which returns the image as it is.

<img width="1013" alt="Screenshot 2022-05-25 at 10 46 02 AM" src="https://user-images.githubusercontent.com/266324/170185163-07482693-640c-43a6-a121-3f3795256207.png">

Also, can you please tell me why we are updating `id` for the `img` class name? The reason I'm asking is that if we use an image block, the same class is used for the block wrapper `div` as well. So, is updating image class necessary?

Reply:
@akshitsethi Can you please send me steps to reproduce that?

I replace the id because when a new image is imported into WordPress it gets a new ID for the attachment and it was not updated in the IMG tag class.

Reply:
This code only replace the_content IMG tag only not other dynamic blocks.

Reply:
@mukeshpanchal27 The class added to the wrapper for the image block was done manually, so my comment above can be ignored. Still, below is the screenshot to explain what I meant. This is from the image block post of the Theme_Unit_Test data.

<img width="1382" alt="Screenshot 2022-05-25 at 1 33 54 PM" src="https://user-images.githubusercontent.com/266324/170214353-ef90cbe0-9958-4b0c-93cc-af98d1d11451.png">


Overall, this looks good to me. Can we address the edge case like the one I'm experiencing where the `updated_attachment_id` is coming out to be 0. Something like below:

```
if ( ! $updated_attachment_id ) {
	$updated_attachment_id = $attachment_id;
}
```

What are your thoughts on this?

Reply:
I don't think we have to check it because if the current attachment ID and the static content image are not the same ID then it will not affect anything.

`if ( $attachment_id !== $updated_attachment_id ) {`

Reply:
This is for the case which I'm experiencing. Please see the two screenshots below:

**On this branch**
<img width="1269" alt="Screenshot 2022-05-25 at 2 25 57 PM" src="https://user-images.githubusercontent.com/266324/170223899-e84a2b2b-59f0-4426-bad8-2f234139eb3b.png">


**On trunk**
<img width="1236" alt="Screenshot 2022-05-25 at 2 26 17 PM" src="https://user-images.githubusercontent.com/266324/170223939-b79695c2-07a7-470c-9642-e865951414b6.png">

These are the same images but when on this branch it only updates the first image to `webp` and the rest of the other images on page are not updated. When I checked the value of `updated_attachment_id` in the loop, it only existed for the first image and rest others were `0`.

This is weird because the `webp` version does exist but still this needs to be investigated further as to why this is happending because on trunk it is working correctly. I'll take out some time today to check this out and update you.